index.js

```diff
@@ -21,6 +23,30 @@ const transformBabel = src => {
   return result;
 };
 
+
+const transformTs = (src, path) => {
+  const {config} = tsconfig.loadSync(process.cwd());
+  let result;
+  try {
+    result = tsc.transpile(
+      src,
+      config.compilerOptions,
+      path,
+      []
+    );
+  } catch (error) {
+    // eslint-disable-next-line
+    console.error('Failed to compile src with `tsc` at `vue-preprocessor`');
+  }
+  return result;
+};
+
+const transforms = {
+  'ts': transformTs,
```

Comment:
just a nitpick, but you don't have to write keys as string, that is done by javascript under the hood